src/Core/Managed/Shared/Extensibility/Metrics/MetricTerms.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+
+namespace Microsoft.ApplicationInsights.Extensibility.Metrics
+{
+    internal static class MetricTerms
+    {
+        private const string MetricPropertiesNamePrefix = "Microsoft.ApplicationInsights.Metrics";
+        public static class Extraction
+        {
+            public static class ConsideredByProcessors
+            {
+                public static class Moniker
+                {
+                    public const string Key = MetricPropertiesNamePrefix + ".Extraction.ConsideredByProcessors";
+                }
+            }
+        }
+
+        public static class Autocollection
+        {
+            public static class Moniker
+            {
+                public const string Key = MetricPropertiesNamePrefix + ".MetricIsAutocollected";
+                public const string Value = "True";
+
+            }
+
+            public static class MetricNames
+            {
+                public static class Request
+                {
+                    public const string Duration = "Server response time";
+                }
+
+                public static class DependencyCall
+                {
+                    public const string Duration = "Dependency duration";
+                }
+            }
+
+            public static class Request
+            {
+                public static class PropertyName
+                {
+                    public const string Success = "Success";
+                }
+            }
+
+            public static class DependencyCall
+            {
+                public static class PropertyName
+                {
+                    public const string Success = "Success";
+                    public const string TypeName = "Type";
+                }
+
+                public static class TypeName
```

Comment:
Hi @dnduffy ,

I have actually seen the current approach in the past and have used it myself.
I am not sure what you mean, so let m see if I understand you correctly by making an example.
Are you looking for this?:
---
namespace Microsoft.ApplicationInsights.Extensibility.Metrics.MetricTerms
{
    internal static class Shared
    {
        public const string MetricPropertiesNamePrefix = "Microsoft.ApplicationInsights.Metrics";
    }

    namespace Extraction.ProcessedByExtractors
    {
        internal static class Moniker
        {
            public const string Key = Shared.MetricPropertiesNamePrefix + ".Extraction.ProcessedByExtractors";
        }
    }

    namespace Autocollection
    {
        internal static class Moniker
        {
            public const string Key = Shared.MetricPropertiesNamePrefix + ".MetricIsAutocollected";
            public const string Value = "True";
        }

        namespace MetricNames
        {
            internal static class Request
            {
                public const string Duration = "Server response time";
            }

            internal static class DependencyCall
            {
                public const string Duration = "Dependency duration";
            }
        }

        namespace Request
        {
            internal static class PropertyName
            {
                public const string Success = "Success";
            }
        }

        namespace DependencyCall
        {
            internal static class PropertyName
            {
                public const string Success = "Success";
                public const string TypeName = "Type";
            }

            internal static class TypeName
            {
                public const string Other = "Other";
            }
        }
    }
}
---
If yes, then that approach has a number of problems:
- Namespaces are not a proper encapsulation. Default tools will tend to "use" then ratehr than typing out everything from MetricTerms... This will result in short identifiers like Request.Duration in code, and so - defeat the purpose.
- It is not clear that even if the user was to type out a longer ID, then MetricTerms is the part to start while the outer components should be "used".
- Intellisense will begin showing empty nonsensical namespaces to people who reference the assembly.
- The convention of correspondence between folder structure and namespaces gets broken.

All in all, if I understood your proposal correctly, then I do not think it is a better alternative. I am not sure why you have concerns with the current approach. It plays very well together with scoping in general and with how intellisense works.